notebook-lm-clone/src/generation/rag.py

```
@@ -0,0 +1,281 @@
+            # Step 5: Create result object
+            rag_result = RAGResult(
+                query=query,
+                response=response,
+                sources_used=sources_info,
+                retrieval_count=len(search_results)
+            )
+            
+            logger.info(f"Response generated successfully using {len(sources_info)} sources")
+            return rag_result
+            
+        except Exception as e:
```

Ruff (0.13.1)

Do not catch blind exception: `Exception`

(BLE001)